src/plone/volto/testing.py

```diff
@@ -1,111 +1,121 @@
-from plone import api
 from plone.app.contenttypes.testing import PLONE_APP_CONTENTTYPES_FIXTURE
 from plone.app.robotframework.testing import REMOTE_LIBRARY_BUNDLE_FIXTURE
 from plone.app.testing import applyProfile
 from plone.app.testing import FunctionalTesting
 from plone.app.testing import IntegrationTesting
-from plone.app.testing import login
-from plone.app.testing import logout
 from plone.app.testing import PloneSandboxLayer
 from plone.app.testing import setRoles
 from plone.app.testing import TEST_USER_ID
-from plone.app.testing import TEST_USER_NAME
+from plone.distribution.testing.layer import PloneDistributionFixture
 from plone.testing import z2
 
-import plone.volto
-import plone.volto.coresandbox
-
-
-try:
-    from Products.CMFPlone.factory import PLONE60MARKER
-
-    PLONE60MARKER  # pyflakes
-except ImportError:
-    PLONE_6 = False
-else:
-    PLONE_6 = True
+import plone.app.caching  # noQA
+import plone.app.discussion  # noQA
+import plone.app.iterate  # noQA
+import plone.app.multilingual  # noQA
+import plone.app.upgrade  # noQA
+import plone.volto  # noQA
+import plone.volto.coresandbox  # noQA
+import Products.CMFPlacefulWorkflow  # noQA
+
+
+ANSWERS = {
+    "site_id": "plone",
+    "title": "Plone Site",
+    "description": "A Plone Site with Volto",
+    "site_logo": "name=teste;data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg==",  # noQA
+    "default_language": "en",
+    "portal_timezone": "America/Sao_Paulo",
+    "setup_content": True,
+}
+
+
+class BaseFixture(PloneDistributionFixture):
+    PACKAGE_NAME = "plone.volto"
+    SITES = (("default", ANSWERS),)
+    _distribution_products = (
+        ("plone.app.contenttypes", {"loadZCML": True}),
+        ("plone.app.caching", {"loadZCML": True}),
+        ("plone.app.iterate", {"loadZCML": True}),
+        ("plone.app.multilingual", {"loadZCML": True}),
+        ("plone.app.upgrade", {"loadZCML": True}),
+        ("Products.CMFPlacefulWorkflow", {"loadZCML": True}),
+        ("plone.restapi", {"loadZCML": True}),
+        ("plone.distribution", {"loadZCML": True}),
+    )
+
+
+BASE_FIXTURE = BaseFixture()
 
 
 class PloneVoltoCoreLayer(PloneSandboxLayer):
 
-    defaultBases = (PLONE_APP_CONTENTTYPES_FIXTURE,)
-
-    def setUpZope(self, app, configurationContext):
-        # Load any other ZCML that is required for your tests.
-        # The z3c.autoinclude feature is disabled in the Plone fixture base
-        # layer.
-        self.loadZCML(package=plone.volto)
-
-    def setUpPloneSite(self, portal):
-        setRoles(portal, TEST_USER_ID, ["Manager"])
-        login(portal, TEST_USER_NAME)
-        api.content.create(
-            type="Document", id="front-page", title="Welcome", container=portal
-        )
-        logout()
-        applyProfile(portal, "plone.volto:default")
+    defaultBases = (BASE_FIXTURE,)
 
 
-PLONE_VOLTO_CORE_FIXTURE = PloneVoltoCoreLayer()
```

Comment:
Please don't rename these fixtures. It's a breaking change for any package that uses them.